File: Dante1/Module1.02/Zadanie2.13/main.c
#include <stdio.h>

int main()
{
    float num1, num2, num3, num4, num5;

    printf("Enter:\n");

    int check = scanf("%f %f %f %f %f",&num1, &num2, &num3, &num4, &num5);
    if(check != 5)
    {
        printf("Incorrect input");
        return 1;
    }

    float sum = num1 + num2 + num3 + num4 + num5;
    float middle = sum / 5;

    float max = num1;
    if (num2 > max)
    {
        max = num2;
    }
    if (num3 > max)
    {
        max = num3;
    }
    if (num4 > max)
    {
        max = num4;
    }
    if (num5 > max)
    {
        max = num5;
    }

    float min = num1;
    if (num2 < min)
    {
        min = num2;
    }
    if (num3 < min)
    {
        min = num3;
    }
    if (num4 < min)
    {
        min = num4;
    }
    if (num5 < min)
    {
        min = num5;
    }

    printf("\n%.2f\n%.2f\n%.2f",min,max,middle);
    return 0;
}

File: Dante1/Module1.02/Zadanie2.13/main.zig
const std = @import("std");

pub fn main() !void {
    var stdin = std.io.getstdin().reader();
    var stdout = std.io.getstdout().writer();

    try stdout.print("Enter 5 numbers: ", .{});

    var buffer: [100]u8 = undefined;
    const line = try stdin.readUntilDelimiterOrEof(&buffer, '\n');

    var iter = std.mem.tokenizeAny(u8, line.?, " \t");
    var nums: [5]f32 = undefined;

    var i: usize = 0;
    while (iter.next()) |tok| : (i += 1) {
        if (i >= nums.len) break;
        nums[i] = try std.fmt.parseFloat(f32, tok);
    }

    const sum: f32 = nums[0] + nums[1] + nums[2] + nums[3] + nums[4];
    const middle: f32 = sum / 5.0;

    var max: f32 = nums[0];
    var min: f32 = nums[0];

    if (nums[1] > max) max = nums[1];
    if (nums[2] > max) max = nums[2];
    if (nums[3] > max) max = nums[3];
    if (nums[4] > max) max = nums[4];

    if (nums[1] < min) min = nums[1];
    if (nums[2] < min) min = nums[2];
    if (nums[3] < min) min = nums[3];
    if (nums[4] < min) min = nums[4];

    try stdout.print("Result: min = {}, max = {}, middle = {}\n", .{ min, max, middle });
}
